### PR TITLE
Initialize all variables spinningbody

### PR DESCRIPTION
**Pull request by patkenneally**
* **Tickets addressed:** NA
* **Review:** By commit 
* **Merge strategy:** Merge (no squash)

## Description
This PR initializes all values of the class. Vectors are initialized using class initialization, while matrices are initialized in the constructor using `setIdentity()`  for brevity. 

## Verification
CI runs. I need to think about what additional testing could be done on this module.

## Documentation
NA

## Future work
NA

### CODE REVIEW

File: src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.h [hunk 1]
```diff
@@ -51,12 +51,12 @@ class SpinningBodyTwoDOFStateEffector: public StateEffector, public SysModel {
...
-    Eigen::Vector3d r_Sc2S2_S2;         //!< [m] vector pointing from upper spinning frame S2 origin to point Sc2 (center of mass of the upper spinner) in S2 frame components
-    Eigen::Vector3d s1Hat_S1;           //!< -- first spinning axis in S1 frame components.
-    Eigen::Vector3d s2Hat_S2;           //!< -- second spinning axis in S2 frame components.
+    Eigen::Vector3d r_S1B_B{0.0,0.0,0.0};            //!< [m] vector pointing from body frame B origin to lower spinning frame S1 origin in B frame components
+    Eigen::Vector3d r_S2S1_S1{0.0,0.0,0.0};          //!< [m] vector pointing from lower spinning frame S1 origin to upper spinning frame S2 origin in S1 frame components
+    Eigen::Vector3d r_Sc1S1_S1{0.0,0.0,0.0};         //!< [m] vector pointing from lower spinning frame S1 origin to point Sc1 (center of mass of the lower spinner) in S1 frame components
+    Eigen::Vector3d r_Sc2S2_S2{0.0,0.0,0.0};         //!< [m] vector pointing from upper spinning frame S2 origin to point Sc2 (center of mass of the upper spinner) in S2 frame components
+    Eigen::Vector3d s1Hat_S1{0.0,0.0,0.0};           //!< -- first spinning axis in S1 frame components.
```
Comment (joaogvcarneiro) [inline src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.h]: Don't we want to initialize the unit vectors to ``[1,0,0]`` or something?

File: src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.h [hunk 2]
```diff
@@ -106,34 +106,34 @@ class SpinningBodyTwoDOFStateEffector: public StateEffector, public SysModel {
...
-    Eigen::Vector3d rDot_Sc1B_B;        //!< [m/s] inertial frame time derivative of r_Sc1B_B
-    Eigen::Vector3d rDot_Sc2B_B;        //!< [m/s] inertial frame time derivative of r_Sc2B_B
-    Eigen::Vector3d omega_S1B_B;        //!< [rad/s] angular velocity of the S1 frame wrt the B frame in B frame components
-    Eigen::Vector3d omega_S2S1_B;       //!< [rad/s] angular velocity of the S2 frame wrt the S1 frame in B frame components
-    Eigen::Vector3d omega_S2B_B;        //!< [rad/s] angular velocity of the S2 frame wrt the B frame in B frame components
-    Eigen::Vector3d omega_BN_B;         //!< [rad/s] angular velocity of the B frame wrt the N frame in B frame components
-    Eigen::Vector3d omega_S1N_B;        //!< [rad/s] angular velocity of the S1 frame wrt the N frame in B frame components
-    Eigen::Vector3d omega_S2N_B;        //!< [rad/s] angular velocity of the S2 frame wrt the N frame in B frame components
-    Eigen::MRPd sigma_BN;               //!< -- body frame attitude wrt to the N frame in MRPs
+    Eigen::Vector3d s1Hat_B{0.0,0.0,0.0};            //!< -- first spinning axis in B frame components
```
Comment (joaogvcarneiro) [inline src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.h]: See the above comment.

File: src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.cpp [hunk 3]
```diff
@@ -37,6 +37,20 @@ SpinningBodyTwoDOFStateEffector::SpinningBodyTwoDOFStateEffector()
     this->IS2PntSc2_S2.setIdentity();
     this->dcm_S10B.setIdentity();
     this->dcm_S20S1.setIdentity();
+    this->ATheta.setIdentity();
```
Comment (joaogvcarneiro) [inline src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.cpp]: Here, I think we can use ``setZero()`` and leave ``setIdentity()`` to the DCMs and inertias (lines 47-51).

**Review by joaogvcarneiro — CHANGES_REQUESTED** [review]
Thanks for this PR. I just have a couple of comments, but nothing major.

**Review by thibaudteil — APPROVED** [review]

File: src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.h (hunk 1 above)
Comment (patkenneally) [reply src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.h]: Good point. Change made!

File: src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.h (hunk 2 above)
Comment (patkenneally) [reply src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.h]: Done.

File: src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.cpp (hunk 3 above)
Comment (patkenneally) [reply src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.cpp]: For sure! Done.

File: src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.cpp [hunk 4]
```diff
@@ -37,6 +37,20 @@ SpinningBodyTwoDOFStateEffector::SpinningBodyTwoDOFStateEffector()
     this->IS2PntSc2_S2.setIdentity();
     this->dcm_S10B.setIdentity();
     this->dcm_S20S1.setIdentity();
+    this->ATheta.setZero();
+    this->BTheta.setZero();
+    this->rTilde_Sc1B_B.setIdentity();
```
Comment (joaogvcarneiro) [inline src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.cpp]: Can you make lines 42-46 and 52-53 use ``setZero()``?

**Review by joaogvcarneiro — APPROVED** [review]
See single comment!

**Review by codyallard28 — APPROVED** [review]

File: src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.cpp (hunk 4 above)
Comment (patkenneally) [reply src/simulation/dynamics/spinningBodies/spinningBodiesTwoDOF/spinningBodyTwoDOFStateEffector.cpp]: Done

**Pull request merged**